talk/objectorientation/adl.tex

```diff
@@ -144,3 +144,27 @@
     \end{columns}
   \end{block}
 \end{frame}
+
+\begin{frame}[fragile]
+  \frametitlecpp[98]{ADL and Hidden Friends}
+  \begin{block}{Hidden friend declarations participate in ADL}
+    \begin{itemize}
+      \item Friends defined in class scope are ``hidden friends''
+      \item They are \emph{not} class members (no access to \mintinline{cpp}{this})
+      \item They participate in ADL, e.g.\ \mintinline{cpp}{std::cout << complex}
+    \end{itemize}
+  \end{block}
+  \begin{exampleblock}{Ostream operator as hidden friend}
+    \begin{cppcode*}{gobble=2,linenos=false}
```

Comment:
What's wrong with line numbers here? I like them, so I can refer to them when talking.

Reply:
The stuff didn't fit on the slide. I'll try again.